client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+		col_date.set_sort_column_id(6)
+		self.treeview_local.append_column(col)
+		self.treeview_local.append_column(col_perm)
+		self.treeview_local.append_column(col_size)
+		self.treeview_local.append_column(col_date)
+
+		self.treeview_local.connect('row-expanded', self.expand_row)
+		self.treeview_local.connect('row-collapsed', self.collapse_row)
+		self._tvmodel = Gtk.TreeStore(str, Pixbuf, str, str, str, int, str)
+
+		self.local_hidden = True
+		self.treeview_local.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
+		menu_item = Gtk.CheckMenuItem.new_with_label('Show Hidden Files')
+		menu_item.connect('toggled', self.hidden_files)
+		self.hidden_files = menu_item
+		self.popup_menu.append(menu_item)
+
+		menu_item = Gtk.MenuItem.new_with_label('Delete')
+		menu_item.connect('activate', self.delete_prompt)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == 3:
+			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
+			return True
+		return
+
+	def expand_row(self, _, treeiter, treepath):
+		newPath = self._tvmodel[treeiter][2]  # pylint: disable=unsubscriptable-object
+		self.load_dirs(newPath, treeiter)
+		self._tvmodel.remove(self._tvmodel.iter_children(treeiter))
+
+	def collapse_row(self, _, treeiter, treepath):
+		current = self._tvmodel.iter_children(treeiter)
+		while current:
+			self._tvmodel.remove(current)
+			current = self._tvmodel.iter_children(treeiter)
+		self._tvmodel.append(treeiter, [None, None, None, None, None, None, None])
+
+	def hidden_files(self, _):  # pylint: disable=method-hidden
+		self.local_hidden = not self.local_hidden
+		self.refresh()
+
+	def refresh(self):
+		model = self._tvmodel
+		exp_lines = []
+		model.foreach(lambda model, path, iter: exp_lines.append(path) if self.treeview_local.row_expanded(path) else 0)
+		self.treeview_local.collapse_all()
+		for path in exp_lines:
+			self.treeview_local.expand_row(path, False)
+
+	def delete_prompt(self, _):
+		selection = self.treeview_local.get_selection()
+		model, treeiter = selection.get_selected()
+		dialog = Gtk.Dialog('Warning')
+		label = 'Are you sure\n you want to delete this {0}?\n'.format('directory' if model[treeiter][5] == -1 else 'file')
+		label = Gtk.Label(label)
+		label.set_justify(Gtk.Justification.CENTER)
+		dialog.vbox.pack_start(label, True, True, 0)
+		label.show()
+		dialog.add_button(Gtk.STOCK_CANCEL, Gtk.ResponseType.REJECT)
+		dialog.add_button(Gtk.STOCK_OK, Gtk.ResponseType.ACCEPT)
+		response_id = dialog.run()
+		dialog.destroy()
+		if response_id == -3:
+			self.delete(model, treeiter)
+
+class LocalDirectory(DirectoryBase):
+	def __init__(self, treeview, logger, upload_button, *args, **kwargs):
+		super(LocalDirectory, self).__init__(treeview, logger)
+		self.upload_button = upload_button
+		self.load_dirs(os.path.abspath(os.sep))
```

Comment:
This should be set from an argument, and it should probably default to the current working directory.